crates/holochain/src/conductor/api/api_external/app_interface.rs

```diff
@@ -81,22 +79,33 @@ impl AppInterfaceApi {
                     .get_app_info(&installed_app_id)
                     .await?,
             )),
-            AppRequest::CallZome(signed_zome_call) => {
-                let zome_call_params = signed_zome_call
+            AppRequest::CallZome(zome_call_params_signed) => {
+                let zome_call_params = zome_call_params_signed
                     .bytes
                     .clone()
                     .decode::<ZomeCallParams>()
                     .map_err(|e| ConductorApiError::SerializationError(e.into()))?;
-                let zome_call = ZomeCall {
-                    signed: *signed_zome_call,
-                    params: zome_call_params,
-                };
-                match self.conductor_handle.call_zome(zome_call.clone()).await? {
+                if !is_valid_signature(
```

Comment:
After deserializing the parameters, verify signature of incoming zome calls. This is one of the two points where a network call comes in and authentication is required.

Reply:
Nice. 

Slightly unrelated to this change. It makes the conductor code rather hard to re-use when logic is placed in the AppRequest::* match block. As much as possible, please can the payload from the interface call be passed to a conductor method? I think this is just about justified to ask for on this PR because it was the case previously that the payload was passed along. 

This is useful if you're wanting to call conductor code directly by using Holochain as a library and it would also make it easier to build an interface other than websockets, like HTTP

Reply:
I see what you mean. I'll move it to something like `conductor.handle_external_zome_call` so that it's clear this only needs to happen for zome calls coming in from somewhere outside of the conductor.

Reply:
Even though it seems to logically belong into `holochain/src/conductor/api/api_external`, I'll add it to `conductor/conductor` as you request.